ecommerce/frontend/src/components/pages/VendorHomepage.js

```diff
@@ -0,0 +1,137 @@
+import { useAppContext } from '../../context/AppContext'
+import React, { useEffect, useState } from 'react'
+import { Spin, Button } from 'antd'
+import { VendorPageContent } from '../VendorContent/VendorPageContent'
+import { formatProduct, formatSearchQueryParams } from '../../utils'
+import { VendorSplash } from '../VendorContent/VendorSplash'
+import { VendorReviews } from '../VendorContent/VendorReviews'
+import { api } from '../../api'
+import { Tabs } from 'antd'
+import './VendorHomepage.less'
+import { PlusCircleOutlined } from '@ant-design/icons'
+import { ProductModal } from '../product_modal/ProductModal'
+
+const { TabPane } = Tabs
+
+export const VendorHomepage = props => {
+    const { vendorId } = props.match.params
+    const { user } = useAppContext()
+    const isVendorAndOwner = user.type === 'vendor' && vendorId === user.id.toString()
+    const [editMode, setEditMode] = useState(isVendorAndOwner)
+    const [modalVisibility, setModalVisibility] = useState(false)
+    const [modalType, setModalType] = useState('add')
+    const [modalProduct, setModalProduct] = useState(undefined)
+
+    const onDeleteProductCard = productId => {
+        console.log('delete product card', productId)
+
+        // async function fetch() {
+        //     try {
+        //         const { data } = await api.delete(`/vendor/product`, { id: productId })
+        //     } catch (error) {
+        //         console.error('failed to delete product', error)
+        //     } finally {
+        //     }
+        // }
+
+        // fetch()
```

Comment:
should this be commented?

Reply:
Commented out, just because I don't have a legit Vendor account with its products. It gets 500 error, I think because of invalid token usage